sdks/python/apache_beam/metrics/metric.py

```diff
@@ -130,10 +130,16 @@ def _matches_name(filter, metric_key):
     if not filter.names and not filter.namespaces:
       return True
 
-    if ((filter.namespaces and metric_key.metric.namespace in filter.namespaces)
-        or (filter.names and metric_key.metric.name in filter.names)):
-      return True
-    return False
+    if filter.names and filter.namespaces:
```

Comment:
I think the condition is probably better expressed as
not ((filter.namespaces and metric_key.metric.namespace not in filter.namespaces) or (filter.names and metric_key.metric.name not in filter.names))